### PR TITLE
Migrate kafkametricsreceiver to the new Metrics Builder

### PR DESCRIPTION
**Pull request by abhide**
**Description:** <Describe what has changed.>

**Link to tracking Issue:** #7142 

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: receiver/kafkametricsreceiver/topic_scraper.go [hunk 1]
```diff
@@ -82,35 +83,37 @@ func (s *topicScraper) scrape(context.Context) (pdata.Metrics, error) {
 		}
 		labels := pdata.NewAttributeMap()
```
Comment (dmitryax) [inline receiver/kafkametricsreceiver/topic_scraper.go:84]: This variable can be removed now?

Comment (abhide) [reply receiver/kafkametricsreceiver/topic_scraper.go]: Yes, this is unused. Removed it from consumer_scraper.go as well.
Thanks for catching this.

File: receiver/kafkametricsreceiver/consumer_scraper.go [hunk 2]
```diff
@@ -143,15 +142,13 @@ func (s *consumerScraper) scrape(context.Context) (pdata.Metrics, error) {
...
-			labels.UpsertString(metadata.A.Topic, topic)
 			if isConsumed {
 				var lagSum int64
 				var offsetSum int64
 				for partition, block := range partitions {
-					labels.UpsertInt(metadata.A.Partition, int64(partition))
 					consumerOffset := block.Offset
 					offsetSum += consumerOffset
-					addIntGauge(ilm.Metrics(), metadata.M.KafkaConsumerGroupOffset.Name(), now, labels, consumerOffset)
+					s.mb.RecordKafkaConsumerGroupOffsetDataPoint(now, consumerOffset, group.GroupId, topic, fmt.Sprintf("%d", partition))
```
Comment (dmitryax) [inline receiver/kafkametricsreceiver/consumer_scraper.go:151]: Hm this look like a breaking change caused by metrics builder limitation. Even if we potentially can accept this particular int->string change, I believe we should enhance the metrics builder to allow attributes of different types.

Do you think you can add this improvement to the metrics builder before merging this PR?

Comment (dmitryax) [reply receiver/kafkametricsreceiver/consumer_scraper.go:151]: Submitted an issue https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/8127

Comment (abhide) [reply receiver/kafkametricsreceiver/consumer_scraper.go:151]: Hey @dmitryax I will work on changing metrics builder. 
Do we want to merge this PR as-is and then fix this as part of #8127? 
I am fine either way.

Comment (dmitryax) [reply receiver/kafkametricsreceiver/consumer_scraper.go:151]: I would prefer avoid introducing breaking changes, so i think we should rather hold off on this one until https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/8127 is resolved

**Comment by abhide** [comment]
@dmitryax can you please re-open this PR? I don't a button to reopen this PR; may be because I don't have enough access.

**Comment by dmitryax** [comment]
Reopened. Please rebase

**Comment by dmitryax** [comment]
@abhide I've reopen it again. Let us know if you are you working on this

**Comment by abhide** [comment]
> @abhide I've reopen it again. Let us know if you are you working on this

Thanks @dmitryax for re-opening this. I will post a new patch to this PR.

File: receiver/kafkametricsreceiver/broker_scraper.go [hunk 3]
```diff
@@ -20,7 +20,7 @@ import (
 	"time"
 
 	"github.com/Shopify/sarama"
-	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/model/pdata"
```
Comment (dmitryax) [inline receiver/kafkametricsreceiver/broker_scraper.go:23]: This module is not available anymore. Please use `go.opentelemetry.io/collector/pdata` instead

File: receiver/kafkametricsreceiver/broker_scraper.go [hunk 4]
```diff
@@ -57,11 +58,12 @@ func (s *brokerScraper) scrape(context.Context) (pmetric.Metrics, error) {
 
 	brokers := s.client.Brokers()
 
-	md := pmetric.NewMetrics()
-	ilm := md.ResourceMetrics().AppendEmpty().ScopeMetrics().AppendEmpty()
-	ilm.Scope().SetName(instrumentationLibName)
-	addIntGauge(ilm.Metrics(), metadata.M.KafkaBrokers.Name(), pcommon.NewTimestampFromTime(time.Now()), pcommon.NewMap(), int64(len(brokers)))
+	md := pdata.NewMetrics()
```
Comment (dmitryax) [inline receiver/kafkametricsreceiver/broker_scraper.go:61]: Please revert this change. IT should be `pmetric`, `pdata` module is not available anymore

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+### 💡 Enhancements 💡
+
+- `kafkametricsreceiver`: Migrate kafkametricsreceiver to the mdatagen metrics builder (#7142)
```
Comment (dmitryax) [inline CHANGELOG.md:7]: This changelog line should be migrated to the new process https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#changelog

**Review by dmitryax — COMMENTED** [review]
Hey @abhide, please let us know if you have time to update the PR. If not, we can take care of it.

**Comment by abhide** [comment]
> Hey @abhide, please let us know if you have time to update the PR. If not, we can take care of it.

@dmitryax Apologies, I have been missing in action for the past 2 months. I will take a look at this PR this week and address your review comments.

**Comment by abhide** [comment]
@dmitryax Haven't had the time to complete this PR. 
I see you are working on https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/13319; at this point lets abandon this PR in favor the one you are working on.

**Comment by dmitryax** [comment]
Hi @abhide. Sorry, I just wanted to close the story. It's the last receiver to migrate to be able to remove the old mdatagen. Let me know if you want to take something else. There are a lot of things to work on in this project:)
